enhancements/cli-in-cluster-management.md

```diff
@@ -0,0 +1,172 @@
+of its deliverable binary.  An in-cluster CLI manager controller will access the CRs to populate and maintain an in-cluster Krew Index.  Users will install OpenShift tools that are known compatible
+with each cluster version through `oc krew`.    
+
+A CLI must provide a ClusterCLI CR.  The result of this proposal will be:
+* ClusterCLI Custom Resource Definition
+* Krew Index and a controller to create the index from CLI CRs.
+* mechanism to manage CLIs via ClusterCLI CRs and Krew
+* a reference consumer implementation, oc, that other CLIs can follow
+
+The CLI in-cluster management will require:
+- CLI images in an available registry- each CLI will produce an image to deliver its artifacts
+- Consumers - through an in-cluster Krew Index and the krew command, consumers will manage OpenShift CLIs
+  Consumers will require access to CLI images.  In disconnected environments, this will be a mirrored local registry. 
+- Producers - each CLI that wants to offer its binaries in-cluster is responsible for providing its metadata and description via a ClusterCLI as well as
+  a CLI image to deliver its artifacts. 
+
+### Non-Goals
+
+* CLI manager will not build or serve the binaries.  It will know where to find them.
+* This proposal is not concerned with _which_ binaries will be managed.  This proposal is meant to determine the mechanism only.  Consumers and publishers are clients of the mechanism. 
+* CLI manager will not create or update the ClusterCLI Custom Resources, those will be managed by individual CLIs and their operators. 
+
+The ClusterCLI CRD will be installed either by a new lightweight operator or by an existing operator.
+The ClusterCLI/oc Custom Resource will serve as a reference consumer implementation for other CLIs to follow.    
+
+[for an example I've created a ClusterCLI controller that will populate 3 test CRs, oc, odo, and kn](https://github.com/sallyom/clustercli)
+
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI in-cluster manager for various CLIs available for OpenShift and related services so that I can discover, install and upgrade them
+
+##### example:
+```
+$ oc krew list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+openshift-installer  OpenShift Installer CLI     4.4          4.3
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+
+$ oc krew install odo
+$ oc krew install kn
+$ oc krew update oc
+$ oc krew remove kn
+```
+
+## References
+- [Krew: kubectl plugin manager](https://github.com/GoogleContainerTools/krew/) - manages kubectl plugins from 
```

Comment:
Change link to https://github.com/kubernetes-sigs/krew

Reply:
done